kotlinx-coroutines-core/jvm/test/channels/RendezvousChannelStressLFTest.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.channels
+
+import kotlinx.atomicfu.*
+import kotlinx.coroutines.*
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.AtomicLongArray
+import kotlin.math.*
+import kotlin.test.*
+
+/**
+ * Tests lock-freedom of send and receive operations on rendezvous channel.
+ * There is a single channel with two sender and two receiver threads.
+ * When one sender or receiver gets suspended at most one other operation is allowed to cease having progress
+ * (`allowSuspendedThreads = 1`).
+ *
+ * **Note**: In the current implementation buffered channels are not lock-free, so this test would fail
+ * *         if channel is created with a buffer.
+ */
+class RendezvousChannelStressLFTest : TestBase() {
```

Comment:
Renamed to `ChannelLFStressTest` to have a nice "*StressTest" suffix for local runs and to have LF tested on JDK 6